features/nanostack/FEATURE_NANOSTACK/mbed-mesh-api/source/ThreadInterface.cpp

```diff
@@ -39,7 +39,12 @@ int ThreadInterface::connect()
     // Release mutex before blocking
     nanostack_unlock();
 
+#if (true == MBED_CONF_MBED_MESH_API_THREAD_USE_STATIC_LINK_CONFIG)
     int32_t count = connect_semaphore.wait(30000);
+#else
+    // static link configuration is not used, wait for commissioning to happen
+    int32_t count = connect_semaphore.wait(osWaitForever);
+#endif
```

Comment:
A new PR will be be created to solve connectivity issue. In a meanwhile agreed with @SeppoTakalo  and @mikter that Thread should wait for connectivity for ever in all cases.